app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -227,6 +232,21 @@ class LocationSensorManager : LocationSensorManagerBase() {
                     requestZoneUpdates()
                 }
 
+                val now = System.currentTimeMillis()
+                if (
+                    (!highAccuracyModeEnabled && isBackgroundLocationSetup) &&
+                    ((lastLocationReceived + (DEFAULT_LOCATION_MAX_WAIT_TIME * 2L)) < now)
+                ) {
+                    Log.d(TAG, "Background location updates appear to have stopped, restarting location updates")
+                    removeBackgroundUpdateRequests()
+                } else if (
+                    highAccuracyModeEnabled &&
+                    ((lastLocationReceived + (getHighAccuracyModeUpdateInterval().toLong() * 2000L)) < now)
+                ) {
+                    Log.d(TAG, "High accuracy mode appears to have stopped, restarting high accuracy mode")
+                    stopHighAccuracyService()
```

Comment:
Why use stop + start it later instead of using the restart function?

Reply:
because the next line we resetup background updates from the beginning again and re-evaluate everything